### PR TITLE
If location's couch doc is missing, still delete sql doc. 

### PR DESCRIPTION
**Pull request by biyeun**
Also log this abnormality.

@dannyroberts @kaapstorm 
http://manage.dimagi.com/default.asp?236297#1210449
http://manage.dimagi.com/default.asp?236299#1210463

cc @gcapalbo

### CODE REVIEW

File: corehq/apps/locations/views.py [hunk 1]
```diff
@@ -392,10 +396,23 @@ def archive_location(request, domain, loc_id):
...
+        # in a state where the couch doc is deleted and the sql doc still exists.
+        loc = SQLLocation.objects.prefetch_related(
+            'location_type').get(location_id=loc_id)
+        # delete the sql location anyway
+        loc.get_descendants(include_self=True).delete()
+        logger.error(
+            'Location with ID [{}] in domain [{}] was missing its couch doc '
+            'upon deletion. If this is recurring it might be worth checking '
+            'out any couch to postrgres issues.'.format(loc_id, domain)
+        )
```
Comment (gcapalbo) [inline corehq/apps/locations/views.py]: i think it's going to be a bit more complex than this - so basically i think what we want is to replicate all the behavior that `loc.full_delete()` is doing (deleting all descendant couch and sql locations, closing all supply point cases, and unlinking any users assigned to those locations, @esoergel did i missing anything?) but be able to do it even if there couch or sql locations that are missing anywhere in the descendant chain.

Comment (gcapalbo) [inline corehq/apps/locations/views.py]: @esoergel i think what i was envisioning was a scenario where you can handle deleting despite missing location data on both sides (not just couch side)

Comment (esoergel) [inline corehq/apps/locations/views.py]: If SQL data is missing, you're kinda screwed, I think.  SQLLocations handle all the relationships (parentage, etc), and are pretty strict.  All bulk gets use SQL to get the ids and then `iter_docs` to get couch locations, so if you were missing SQLLocations, you'd still have a very hard time finding the couch locations.  This whole thing just  emphasizes how much we really need to get off couch locations.

**Comment by czue** [comment]
image

**Comment by biyeun** [comment]
lol @czue

**Comment by biyeun** [comment]
@gcapalbo @esoergel updated

**Pull request merged**